Client/mods/deathmatch/logic/luadefs/CLuaColShapeDefs.cpp

```diff
@@ -366,3 +368,87 @@ int CLuaColShapeDefs::IsInsideColShape(lua_State* luaVM)
     lua_pushboolean(luaVM, false);
     return 1;
 }
+
+int CLuaColShapeDefs::GetColPolygonHeight(lua_State* luaVM)
+{
+    CClientColShape* pColShape;
+
+    CScriptArgReader argStream(luaVM);
+    argStream.ReadUserData(pColShape);
+
+    if (!argStream.HasErrors())
+    {
+        if (pColShape->GetShapeType() == COLSHAPE_POLYGON)
+        {
+            CClientColPolygon* pColPolygon = static_cast<CClientColPolygon*>(pColShape);
+            float fFloor, fCeil;
+            pColPolygon->GetHeight(fFloor, fCeil);
+            if (fFloor == std::numeric_limits<float>::min())
+                lua_pushboolean(luaVM, false);
+            else
+                lua_pushnumber(luaVM, fFloor);
+
+            if (fCeil == std::numeric_limits<float>::max())
+                lua_pushboolean(luaVM, false);
+            else
+                lua_pushnumber(luaVM, fCeil);
+
+            return 2;
+        }
+        else
+            argStream.SetCustomError("Colshape have to be type: Polygon to use this function!");
+    }
+    else
+        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+
+    lua_pushboolean(luaVM, false);
+    return 1;
+}
+
+int CLuaColShapeDefs::SetColPolygonHeight(lua_State* luaVM)
+{
+    //  bool SetColPolygonHeight ( colshape theColShape, float floor, float ceil )
+    CClientColShape* pColShape;
+    float            fFloor, fCeil;
+
+    CScriptArgReader argStream(luaVM);
+    argStream.ReadUserData(pColShape);
+    if (argStream.NextIsBool())
```

Comment:
Instead of requiring booleans I think you should make use of `math.huge` and `-math.huge`. This can be read as a float using `argStream.ReadNumber(fFloor)`.

You can check to see if `math.huge` has been used by using `std::isinf(fFloor)`. Don't forget to check `fFloor > 0` or `fFloor < 0`.

In `DoHitDetection` I'm not sure if we should use infinity (instead of min/max). Can you give this a try?